src/api-client.ts

```diff
@@ -44,13 +44,31 @@ export abstract class ApiClient {
     return paramString ? `${pathUrl}?${paramString}` : pathUrl;
   }
 
+  fetchEntity(
+    url: string,
+    options?: GoogleAppsScript.URL_Fetch.URLFetchRequestOptions,
+    payload?: Record<string, any>
+  ) {
+    const res = this.fetchResponse(url, options, payload);
+    return res.getContentText() ? JSON.parse(res.getContentText()) : {};
+  }
+
+  fetchBlob(
+    url: string,
+    options?: GoogleAppsScript.URL_Fetch.URLFetchRequestOptions,
+    payload?: Record<string, any>
+  ) {
+    const res = this.fetchResponse(url, options, payload);
+    return res.getBlob();
+  }
+
   /**
    * Make an HTTPS API request using specified auth method (see 'Auth' class)
    */
-  fetchUrl(
+  fetchResponse(
     url: string,
-    options?: GoogleAppsScript.URL_Fetch.URLFetchRequestOptions,
-    payload?: Record<string, any>
+    options: GoogleAppsScript.URL_Fetch.URLFetchRequestOptions | undefined,
```

Comment:
Why are these specifically undefined and not optional anymore?
The result would be the same no? If you don't have options to parse, then you'd call fetchResponse('...', undefined, payload).